security/pkg/k8s/tokenreview/k8sauthn.go

```diff
@@ -83,10 +89,10 @@ func (authn *K8sSvcAcctAuthn) reviewServiceAccountAtK8sAPIServer(targetToken str
 		Kind:       "TokenReview",
 		Spec: specForSaValidationRequest{
 			Token: targetToken,
-			// TODO: Also check for the JWT audiences. This feature is only available on Kubernetes from
-			//  v1.13 and above.
 			// If the audiences are not specified, the api server will use the audience of api server,
 			// which is also the issuer of the jwt.
+			// This feature is only available on Kubernetes v1.13 and above.
+			Audiences: []string{defaultAudience},
```

Comment:
when is this ever overrided? Seems like default is always used?

Reply:
For 1.3, it's hard-coded. We will revisit it post 1.3 (maybe a patch in 1.3.x) to support configurable trust domain. The audience should be Citadel. We won't support Vault with trustworthy JWT for now

Reply:
Ok to have it hard-coded - but hard-code it in the template as well.

Reply:
And I don't think it matters if Vault or something else is used - as long as they check the same constant for audience it'll probably work.

The intent of audience is to make sure a server doesn't accept a token intended for a different server. 

I'm not entirely sure the spifee URL is a proper use for audience - it's intended to identify workloads, not audiences. You could have any SA or workload implementing the cert signing service. 

Maybe a simple string "istio-ca" as hardcoded audience would be better ?

Reply:
@myidpt wdyt? I'm okay with either option (istio-ca or the spiffee one)

Reply:
"istio-ca" is definitely clean, but what if we have multiple Citadels, do we want to have the ability to separate the usage of JWTs sent to them? Especially the SDS flow allows Citadel to be deployed in a separate cluster. I think leveraging the "trust domain", etc. specs in SPIFFE is generally beneficial in this scenario to prevent misuse of the JWT.

Reply:
I've updated the PR to use "istio-ca". PTAL

Reply:
why not hard code in the template then?

Reply:
It's not always "istio-ca". For example, for GoogleCA, this value is different and Istio cannot control it